File: backend/src/main/java/com/example/backend/dto/TripSheetOutputDTO.java
package com.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
public class TripSheetOutputDTO {
    private Long id;
    private String name;
    private Date creationDate;
    private Date lastModifiedDate;
    private Long version;
    private VehicleOutputDTO vehicle;
    private Set<PersonOutputDTO> persons;
    private List<LocationDTO> locations;
    private Long vehicleDeploymentPlanId;
}

File: frontend/src/app/dtos/TripSheetOutput.dto.ts
import {PersonOutputDto} from "./PersonOutput.dto";
import {LocationDto} from "./Location.dto";
import {VehicleOutputDto} from "./VehicleOutput.dto";

export interface TripSheetOutputDto {
    id: number;
    creationDate: Date;
    lastModifiedDate: Date;
    version: number;
    vehicle: VehicleOutputDto;
    persons: PersonOutputDto[];
    locations: LocationDto[];
    vehicleDeploymentPlanId: number;
}